include/interrupts.h

```diff
@@ -39,6 +39,8 @@ void intr_event_remove_handler(intr_handler_t *ih);
 void intr_event_execute_handlers(intr_event_t *ie);
 void run_event_handlers(unsigned irq);
 
+bool during_intr_handler();
+
 #define intr_disable() __extension__ ({ asm("di"); })
```

Comment:
You are right, forbidding the compiler from reordering `intr_disable()` and `intr_enable()` is indeed superior.
This project was meant to teach us something and you certainly helped in that.